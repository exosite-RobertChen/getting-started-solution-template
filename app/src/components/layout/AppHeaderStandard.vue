<template>
  <div>
    <AppDrawer v-model="drawerActive" :branding="branding" />
    <v-app-bar color="primary" app dark class="pr-1">
      <v-app-bar-nav-icon
        class="ml-3"
        data-qa="HeaderDrawerToggleButton"
        @click="drawerActive = !drawerActive"
      />
      <v-toolbar-title class="hidden-xs-only" v-text="branding.title" />
      <v-spacer />
      <v-btn color="accent" class="mr-0" @click="$emit('openClaimDevice')"
        >Claim Device</v-btn
      >
      <v-menu offset-y>
        <template v-slot:activator="{ on }">
          <v-btn
            class="mx-2 text-none"
            data-qa="HeaderMenuActivator"
            color="primary"
            depressed
            v-on="on"
          >
            <span class="hidden-xs-only mr-1" right>{{ userEmail }}</span>
            <v-icon>person</v-icon>
          </v-btn>
        </template>

        <v-list>
          <v-list-item
            key="logout"
            data-qa="HeaderLogoutButton"
            @click="$emit('logout')"
          >
            <v-list-item-title>Log Out</v-list-item-title>
          </v-list-item>
        </v-list>
      </v-menu>
    </v-app-bar>
  </div>
</template>

<script>
import AppDrawer from '@/components/layout/AppDrawer'

export default {
  name: 'AppHeaderStandard',
  components: {
    AppDrawer
  },
  props: {
    branding: {
      type: Object,
      required: true
    },
    userEmail: {
      type: String,
      required: false,
      default: ''
    }
  },
  data() {
    return {
      drawerActive: false
    }
  }
}
</script>
