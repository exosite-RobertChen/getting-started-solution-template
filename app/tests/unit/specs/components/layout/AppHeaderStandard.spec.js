import Vue from 'vue'
import Vuetify from 'vuetify'
import merge from 'lodash.merge'
import { mount } from '@vue/test-utils'
import AppHeaderStandard from '@/components/layout/AppHeaderStandard.vue'
import branding from '@/../vendor/branding.json'

const factory = overrides => {
  const localVue = Vue
  // to avoid 'component not registered' warning
  localVue.component(
    Vue.component('router-link', {
      render: () => `<div><slot /></div>`
    })
  )

  let defaults = {
    localVue,
    vuetify: new Vuetify(),
    propsData: {
      branding,
      userEmail: 'tester@email.com'
    },
    mocks: {
      $route: {
        name: 'DevicesPages'
      }
    }
  }
  merge(defaults, overrides)
  return mount(AppHeaderStandard, defaults)
}

describe('AppHeaderStandard.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = factory()
  })
  afterEach(() => wrapper.destroy())

  test('can be mounted', () => {
    expect(wrapper.contains(AppHeaderStandard)).toBe(true)
  })

  test('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('toggles app drawer on toggle button click', () => {
    const drawerToggle = wrapper.find('[data-qa=HeaderDrawerToggleButton]')
    expect(wrapper.vm.drawerActive).toBe(false)
    drawerToggle.trigger('click')
    expect(wrapper.vm.drawerActive).toBe(true)
    drawerToggle.trigger('click')
    expect(wrapper.vm.drawerActive).toBe(false)
  })
})
