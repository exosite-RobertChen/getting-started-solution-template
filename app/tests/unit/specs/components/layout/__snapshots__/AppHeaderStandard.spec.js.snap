// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeaderStandard.vue matches snapshot 1`] = `
<div>
  <nav
    class="v-navigation-drawer v-navigation-drawer--close v-navigation-drawer--fixed v-navigation-drawer--is-mobile theme--light"
    style="height: 100vh; top: 0px; transform: translateX(-100%); width: 200px;"
  >
    <div
      class="v-navigation-drawer__content"
    >
      <div
        class="row secondary"
        style="height: 100%;"
      >
        <div
          class="col"
        >
          <div
            class="text-center"
          >
            <img
              alt="brandlogo"
              class="my-4 mx-auto"
              data-qa="DynamicImg"
              src="vendor/brandlogo-dark.png"
              style="align-self: center; width: 90%;"
            />
          </div>
           
          <div
            class="v-list v-sheet v-sheet--tile theme--light"
          >
            <!---->
            <!---->
            <div
              class="white-text darken-2 drawer__tile py-1 v-list-item v-list-item--link theme--light"
              tabindex="0"
            >
              <div
                class="v-list-item__action"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate grey--text lighten-2 material-icons theme--light"
                >
                  code
                </i>
              </div>
               
              <div
                class="v-list-item__content"
                style="margin-left: -8px;"
              >
                <div
                  class="v-list-item__title drawer-item-title"
                >
                  Api
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="grey--text mt-auto drawer__company-footer text-center mb-2"
        >
          <span>
             Exosite Â© 2019 
          </span>
        </div>
      </div>
    </div>
    <div
      class="v-navigation-drawer__border"
    />
  </nav>
   
  <header
    class="pr-1 v-sheet v-sheet--tile theme--dark v-toolbar v-app-bar v-app-bar--fixed primary"
    style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <button
        class="v-app-bar__nav-icon ml-3 v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--default"
        data-qa="HeaderDrawerToggleButton"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-menu theme--dark"
          />
        </span>
      </button>
       
      <div
        class="v-toolbar__title hidden-xs-only"
      >
        Product portal
      </div>
       
      <div
        class="spacer"
      />
       
      <button
        class="mr-0 v-btn v-btn--contained theme--dark v-size--default accent"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          Claim Device
        </span>
      </button>
       
      <div
        class="v-menu"
      >
        <div
          class="v-menu__content theme--light "
          role="menu"
          style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
        />
      </div>
      <button
        class="mx-2 text-none v-btn v-btn--depressed theme--dark v-size--default primary"
        data-qa="HeaderMenuActivator"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <span
            class="hidden-xs-only mr-1"
            right=""
          >
            tester@email.com
          </span>
           
          <i
            aria-hidden="true"
            class="v-icon notranslate material-icons theme--dark"
          >
            person
          </i>
        </span>
      </button>
    </div>
  </header>
</div>
`;
